### ISSUE TITLE
NVDA reads _ character which specifies mnemonic in WPF application

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#3475]
**Reported by mwhapples on 2013-08-28 16:20**
When writing a WPF application (at the moment I am creating a client based application so have only tested it for that), NVDA reads the _ character which is used to specify the mnemonic for the control. This is particularly annoying when the mnemonic is not the first letter of the string and so NVDA breaks up the word when speaking.

Follows is an example of XAML code of such a button where the mnemonic is A.
`<Button><Label>C_ancel</Label></Button>`

I would expect the spoken output to be something like: Cancel Button A
Actual spoken output is: C_ancel Button A

Braille output is affected in a similar way.
**Edit:**
Formatting, clarity and added the following summary:

* Using sample application: [WpfMnemonicsApplication.zip](https://github.com/nvaccess/nvda/files/1228485/WpfMnemonicsApplication.zip)
* There are three buttons defined: Cancel, Next, Back.
* The cancel button is an example of the issue: Cancel has a `Label` tag nested in a `Button` tag. This results in "C_ancel" being reported by NVDA
* The next button has a work-around for the issue, adding `AutomationProperties.Name="Next"` to the `Button` tag with the nested `Label`. This results in "Next" being reported by NVDA.
* The back button does NOT nest the `Label` tag in the `Button` tag. This results in "Back" being reported by NVDA.

These workarounds are discussed in https://github.com/nvaccess/nvda/issues/3475#issuecomment-155309140.

**Comment by bhavyashah** [comment]
@feerrenrut Is the test case attached in https://github.com/nvaccess/nvda/issues/3475#issuecomment-155309137 sufficient for successfully reproducing the reported issue?

**Comment by feerrenrut** [comment]
@bhavyashah Yes, this issue can be reproduced with that test case. The test case also includes two work around options. I have updated the description of this issue to make it clearer.

**Comment by Adriani90** [comment]
@feerrenrut in case this is still reproducible in NVDA 2019.1.1, is this an NVDA issue? As far as I understand this seems to be a mistake in how WPF exposes the information. Is this still the case when creating applications with WPF 4.7 which is the last version? Note that WPF is now part of the .netcore and I guess this will become more important in future applications. If this is an issue in WPF, we should definitely report it to Microsoft along with other WPF issues we might come accross.

**Comment by Adriani90** [comment]
cc: @siagupta0202, @anjali-wpf, @Arpitmathur, @dipeshmsft, @Himgoyalmicro, @harshit7962 this is also one issue in WPF which needs some attention, It impacts alot users of screen readers because the labels are not read out clearly by the screen reading voice when such implementation characters liek the underline are exposed as part of the label.

**Comment by seanbudd** [comment]
This needs to be reported to MS (WPF developers)
